src/main/java/fi/aalto/cs/apluscourses/intellij/model/APlusProject.java

```diff
@@ -40,43 +48,38 @@ public LibraryTable getLibraryTable() {
     return LibraryTablesRegistrar.getInstance().getLibraryTable(project);
   }
 
-  @NotNull
-  public Path getLibraryPath(String name) {
-    return  Paths.get("lib", name);
-  }
-
-  @NotNull
-  public Path getModulePath(String name) {
-    return Paths.get(name);
-  }
-
-  public int resolveLibraryState(String name) {
-    return resolveComponentState(getLibraryTable().getLibraryByName(name), getLibraryPath(name));
-  }
-
-  public int resolveModuleState(String name) {
-    return resolveComponentState(getModuleManager().findModuleByName(name), getModulePath(name));
-  }
-
-  private int resolveComponentState(@Nullable Object componentObj, @NotNull Path dirPath) {
+  /**
+   * Returns a state in which the given component should be.
+   * @param component An IntelliJ specific component.
+   * @param <T> Type of the component.
+   * @return Component state.
+   */
+  public <T extends Component & IntelliJComponent<?>> int resolveComponentState(
+      @NotNull T component) {
     /*
-     * Three cases to check for here:
+     * Four cases to check for here:
+     *   0. The component is in the project but the files are missing -> ERROR.
      *   1. The component is in the project, so its state should be INSTALLED.
      *   2. The component is not in the project but the module files are present in the file
      *      system, so its state should be FETCHED.
      *   3. The component files aren't present in the file system (and by extension the component
      *      isn't in the project), so its state should be NOT_INSTALLED.
      */
-    if (componentObj != null) {
-      return Component.INSTALLED;
+    boolean loaded = component.getPlatformObject() != null;
```

Comment:
this is maybe a bit too abstract `component.getPlatformObject()`? :dizzy_face:

Reply:
I agree that the name "platform object" is a bit lame. Could we come up with a better word?